core/Pos/Core/Block/Union/Types.hs

```diff
@@ -30,6 +33,15 @@ type BlockHeader = Either GenesisBlockHeader MainBlockHeader
 -- | Block.
 type Block = Either GenesisBlock MainBlock
 
+-- | Representation of 'Block' passed to a component.
+data ComponentBlock payload =
+    ComponentBlockGenesis (Some IsGenesisHeader)
+    | ComponentBlockMain
+       { bcmHeader  :: !(Some IsMainHeader)
+       , bcmPayload :: !payload }
+
+type UpdateBlock = ComponentBlock UpdatePayload
```

Comment:
`UpdateBlock` should be defined in `cardano-sl-update` package. For instance, in the same place as before.